### PR TITLE
TT-1934: Fix up some issues with polymorphic deserialization

### PR DESCRIPTION
**Pull request by andy-paine**
Based on comments from @willp-bl 
Update dropwizard-jackson to 1.2.2 (1.2.3 contains extra deps
not in our whitelist)
Remove unnecessary aliases
Add typed exception for certificate loading errors

Authors: @andy-paine

### CODE REVIEW

File: security-utils/src/test/java/uk/gov/ida/common/shared/configuration/PublicKeyFileConfigurationTest.java [hunk 1]
```diff
@@ -44,21 +44,21 @@ public void should_loadPublicKeyWhenUsingAliases() throws Exception {
...
         String path = getClass().getClassLoader().getResource("empty_file").getPath();
-        DeserializablePublicKeyConfiguration publicKeyConfiguration = objectMapper.readValue("{\"type\": \"file\", \"certFile\": \"" + path + "\", \"name\": \"someId\"}", DeserializablePublicKeyConfiguration.class);
+        objectMapper.readValue("{\"type\": \"file\", \"cert\": \"" + path + "\", \"name\": \"someId\"}", DeserializablePublicKeyConfiguration.class);
     }
 
     @Test(expected = InvalidDefinitionException.class)
     public void should_ThrowExceptionWhenIncorrectKeySpecified() throws Exception {
         String path = getClass().getClassLoader().getResource("empty_file").getPath();
-        objectMapper.readValue("{\"type\": \"file\", \"certFileFoo\": \"" + path + "\", \"name\": \"someId\"}", DeserializablePublicKeyConfiguration.class);
+        objectMapper.readValue("{\"type\": \"file\", \"cert\": \"" + path + "\", \"name\": \"someId\"}", DeserializablePublicKeyConfiguration.class);
```
Comment (willp-bl) [inline security-utils/src/test/java/uk/gov/ida/common/shared/configuration/PublicKeyFileConfigurationTest.java]: is this testing for an empty file or bad key?

**Comment by willp-bl** [comment]
LGTM, just that one comment on that test

File: security-utils/src/test/java/uk/gov/ida/common/shared/configuration/PublicKeyFileConfigurationTest.java (hunk 1 above)
Comment (andy-paine) [reply security-utils/src/test/java/uk/gov/ida/common/shared/configuration/PublicKeyFileConfigurationTest.java]: Woops - yeah that's supposed to be testing for a bad JSON key. Should be `certFoo` or something. Having tests about crypto keys and JSON keys together is confusing 😞

**Comment by willp-bl** [comment]
LGTM 👍

**Pull request merged**